File: BudgetZeroApp/app/src/main/java/com/example/budgetzeroapp/tool/adapter/BudgetAdapter.java
package com.example.budgetzeroapp.tool.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.appcompat.content.res.AppCompatResources;

import com.example.budgetzeroapp.AppVars;
import com.example.budgetzeroapp.R;
import com.example.budgetzeroapp.tool.item.CategoryItem;

import java.util.List;

public class BudgetAdapter extends ArrayAdapter<CategoryItem> {

    protected int resourceLayout;
    public BudgetAdapter(List<CategoryItem> items) {
        super(AppVars.getContext(), R.layout.category, items);
        resourceLayout = R.layout.category;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        View v = convertView;
        if (v == null) {
            LayoutInflater vi;
            vi = LayoutInflater.from(AppVars.getContext());
            v = vi.inflate(resourceLayout, null);
        }
        CategoryItem p = getItem(position);
        if (p != null) {
            TextView catName = (TextView) v.findViewById(R.id.category_name);
            if (catName != null) catName.setText(p.getName());
            TextView amount = (TextView) v.findViewById(R.id.amount);
            if (amount != null) {
                String mess = p.getTotal() + " €";
                amount.setText(mess);
            }
            ImageView icon = v.findViewById(R.id.icon);
            if (icon != null)
                icon.setImageDrawable(AppCompatResources.getDrawable(AppVars.getContext(), p.getDrawable()));
            TextView budget = (TextView) v.findViewById(R.id.budget);
            if (budget != null) {
                String mess = "Budget : "+p.getBudget() + " €";
                budget.setText(mess);
            }
        }
        return v;
    }
}

File: NoAppNoGain/app/src/main/java/com/example/noappnogain/adapter/BudgetAdapter.kt
package com.example.noappnogain.adapter

import android.app.AlertDialog
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.recyclerview.widget.RecyclerView
import com.example.noappnogain.R
import com.example.noappnogain.model.Budget
import com.example.noappnogain.model.Projet
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import java.util.ArrayList

class BudgetAdapter(private val budgetList: ArrayList<Budget>) :
    RecyclerView.Adapter<BudgetAdapter.MyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {

        val itemView = LayoutInflater.from(parent.context).inflate(
            R.layout.budget_recycler_view,
            parent, false
        )
        return MyViewHolder(itemView)

    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {

        val currentitem = budgetList[position]

        holder.type.text = currentitem.category
        holder.amount.text = currentitem.montant.toString()

        holder.itemView.setOnClickListener { view ->
            var mAuth: FirebaseAuth? = null
            var mUser: FirebaseUser? = null
            var mBudgetDatabase: DatabaseReference? = null
            mAuth = FirebaseAuth.getInstance()
            mUser = mAuth?.currentUser

            if (mAuth!!.currentUser != null) {
                val uid = mUser!!.uid
                mBudgetDatabase =
                    FirebaseDatabase.getInstance().reference.child("BudgetData").child(uid)
            }

            val mydialog = AlertDialog.Builder(view.context)
            val inflater = LayoutInflater.from(view.context)
            val myviewm: View = inflater.inflate(R.layout.modifier_budget, null)
            mydialog.setView(myviewm)
            val edtAmount = myviewm.findViewById<EditText>(R.id.montant_edt)
            val edtCat = myviewm.findViewById<TextView>(R.id.categorie_edt)

            val type = budgetList[position].category
            val post_key = budgetList[position].id
            val amount = budgetList[position].montant
            edtCat.setText(type)
            edtAmount.setText(amount.toString())

            val btnUpdate = myviewm.findViewById<Button>(R.id.btn_upd_Update)
            val dialog = mydialog.create()
            btnUpdate.setOnClickListener(View.OnClickListener {
                val amount = edtAmount.text.toString().trim { it <= ' ' }
                if (TextUtils.isEmpty(amount)) {
                    edtAmount.error
                    Toast.makeText(view.context, "Echec de l'enregistrement...", Toast.LENGTH_SHORT).show()
                    return@OnClickListener
                }
                val myAmount = amount.toInt()
                if (mAuth?.currentUser != null) {
                    val data = Budget(myAmount, type, post_key)
                    mBudgetDatabase?.child(post_key.toString())?.setValue(data)
                    Toast.makeText(view.context, "Enregistrement réussi...", Toast.LENGTH_SHORT).show()
                }
                dialog.dismiss()
            })
            dialog.show()
        }

    }

    override fun getItemCount(): Int {
        return budgetList.size
    }

    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val type: TextView = itemView.findViewById(R.id.type_txt_budget)
        val amount: TextView = itemView.findViewById(R.id.amount_txt_budget)

    }

}